calabash/Classes/FranklyServer/Routes/LPClearTextRoute.m

```diff
@@ -0,0 +1,140 @@
+#if ! __has_feature(objc_arc)
+#warning This file must be compiled with ARC. Use -fobjc-arc flag (or convert project to ARC).
+#endif
+
+#import "LPClearTextRoute.h"
+#import "UIScriptParser.h"
+#import "LPTouchUtils.h"
+#import "LPCocoaLumberjack.h"
+#import "LPJSONUtils.h"
+
+@implementation LPClearTextRoute
+
+- (BOOL) supportsMethod:(NSString *) method atPath:(NSString *) path {
+  return [method isEqualToString:@"GET"];
+}
+
+- (NSDictionary *)failureResponseWithReason:(NSString *)reason {
+    return @{
+        @"outcome" : @"FAILURE",
+        @"reason" : reason,
+        @"details" : @""
+    };
+}
+
+- (NSDictionary *)successResponseWithResult:(id)result {
+    return @{
+       @"outcome" : @"SUCCESS",
+       @"results" : @[result]
+    };
+}
+
+- (id)firstResponder {
+  NSString *query = @"* isFirstResponder:1 index:0";
+  UIScriptParser *parser = [[UIScriptParser alloc] initWithUIScript:query];
+  [parser parse];
+  NSArray *allWindows = [LPTouchUtils applicationWindows];
+  NSArray *results = [parser evalWith:allWindows];
+  if (!results || results.count == 0) {
+    return nil;
+  } else {
+    return results[0];
+  }
+}
+
+- (NSDictionary *)JSONResponseForMethod:(NSString *) method
+                                     URI:(NSString *) path
+                                    data:(NSDictionary *) data {
+  NSString *reason = @"";
+  id target = [self firstResponder];
+  if (!target) {
+    reason = [NSString stringWithFormat:@"Cannot clear text because no view is first responder"];
+    LPLogError(@"%@", reason);
+    return [self failureResponseWithReason:reason];
+  }
+
+  if (![target respondsToSelector:@selector(isFirstResponder)]) {
+    reason = [NSString stringWithFormat:@"Target %@ does not respond to 'isFirstResponder'", target];
+    return [self failureResponseWithReason:reason];
+  }
+
+  id delegate = nil;
+  if ([target respondsToSelector:@selector(delegate)]) {
+    delegate = [target performSelector:@selector(delegate)];
+  }
+
+  BOOL shouldClearText = YES;
+  NSString *existingText = @"";
+  if ([target respondsToSelector:@selector(text)]) {
+    id currentText = [target performSelector:@selector(text)];
+    if (currentText) {
+      existingText = (NSString *)currentText;
+    }
+  }
+
+  NSRange textRange = NSRangeFromString(existingText);
+  if ([target respondsToSelector:@selector(setText:)]) {
+    shouldClearText = [self shouldClearTextInRange:textRange WithDelegate:delegate withTarget:target];
+    if (shouldClearText) {
+      [target performSelector:@selector(setText:) withObject:@""];
```

Comment:
I *think* the correct behavior is to set the text to nil.  In iOS 7, there was change from `""` to `nil` or the other way around.

I think I would prefer using the LPInvoker instead of `performSelector:`.   If you run the static analyzer on this file, does this call generate an warning?